src/get-started/first-java-service.md

```diff
@@ -0,0 +1,605 @@
+Compile the message:
+
+```shell
+mvn generate-sources
+```
+
+---
+
+Next, let's write the transaction method. Navigate to `MyService`
+and add the following method:
+
+<!--codeinclude-->
+[](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/MyService.java)
+inside_block:ci-add-vehicle
+<!--/codeinclude-->
+
+The annotation accepts an integral _transaction ID_: an identifier of
+the transaction that service clients use to invoke it.
+
+The method accepts `AddVehicle` class: an auto-generated class from our
+protobuf definition. It also accepts the `TransactionContext` allowing
+the transaction to access the database.
+
+The transaction implementation uses the service data schema `MySchema`
+to access the service data.
+
+Note that the transaction throws an [`ExecutionException`](java-binding.md#exceptions)
+in case of a precondition failure: in this case, an attempt to add a vehicle
+with an existing ID. When Exonum catches an `ExecutionException`, it rolls back
+any changes made by this transaction and records its execution status
+as erroneous.
+
+Compile the code:
+
+```shell
+mvn compile
+```
+
+#### Transfer the Ownership Transaction
+
+Let's add the second transaction which will change the owner of a vehicle
+in the registry. It needs as its arguments the ID of the vehicle;
+and a new owner.
+
+Navigate to `transactions.proto` and add a message with the arguments:
+
+<!--codeinclude-->
+[](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-messages/src/main/proto/example/vehicle/transactions.proto)
+block:ChangeOwner
+<!--/codeinclude-->
+
+Compile the message:
+
+```shell
+mvn generate-sources
+```
+
+Then navigate to the service `MyService` and add an implementation with
+appropriate constants:
+
+<!--codeinclude-->
+[](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/MyService.java)
+inside_block:ci-change-owner
+<!--/codeinclude-->
+
+This transaction is similar to the first.
+
+Notice how an update of the `owner` field in an existing `Vehicle` value
+is performed. It creates a builder from the existing object with
+`Vehicle.newBuilder(templateVehicle)` method, updates the field, and builds
+a new object.
+
+Compile the code:
+
+```shell
+mvn compile
+```
+
+#### Service Constructor
+
+Finally, we will add a _service constructor_ — a special type of a transaction,
+that is invoked once when the service is instantiated. We will use it
+to populate our registry with some test data.
+
+It is implemented as the `Service#initialize` method. By default, it has
+an empty implementation in the interface, hence it is not yet present
+in `MyService`.
+
+Override the `Service#initialize` with the following implementation:
+
+<!--codeinclude-->
+[](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/MyService.java)
```

Comment:
Nit: This code excerpt is also slightly too wide for the full layout.

Reply:
Thanks, I've reformatted in exonum/exonum-java-binding@06d2691f50bf11d271062194adfd8caa3781c5ae